runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/ExecutableStageDoFnOperator.java

```diff
@@ -248,33 +253,45 @@ private StateRequestHandler getStateRequestHandler(ExecutableStage executableSta
     return StateRequestHandlers.delegateBasedUponType(handlerMap);
   }
 
-  private static class BagUserStateFactory<K extends ByteString, V, W extends BoundedWindow>
+  static class BagUserStateFactory<K extends ByteString, V, W extends BoundedWindow>
       implements StateRequestHandlers.BagUserStateHandlerFactory<K, V, W> {
 
+    /** Upper limit of the number of valid cache tokens to hand out to the SDK. */
+    private static final int MAX_CACHE_SIZE = 100;
```

Comment:
I think it is good for the initial implementation. Do we want to make it configuration based later on?

Reply:
Actually, this will become obsolete if we start to maintain only a single cache token for all user state and one for each side input.